tests/js/spec/views/issueList/overview.spec.jsx

```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
-          isPinned: true,
+      createWrapper({
+        pathname: '/organizations/org-slug/issues/',
+        query: {
+          cursor: '1443575000:0:1',
+          page: 1,
+          environment: [],
+          project: [],
+          query: 'is:unresolved',
+          statsPeriod: '14d',
         },
       });
 
-      wrapper
-        .find('SmartSearchBar textarea')
-        .simulate('change', {target: {value: 'assigned:me level:fatal'}});
-      wrapper.find('SmartSearchBar form').simulate('submit');
-
-      await tick();
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
 
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          query: expect.objectContaining({
-            project: [123],
-            environment: ['prod'],
-            query: 'assigned:me level:fatal',
-          }),
-        })
-      );
+      fireEvent.click(wrapper.getByLabelText('Previous'));
 
-      const newRouter = {
-        ...router,
-        location: {
-          ...router.location,
-          query: {
-            ...router.location.query,
-            project: [123],
-            environment: ['prod'],
-            query: 'assigned:me level:fatal',
-          },
+      // cursor is undefined because "prev" cursor is === initial "next" cursor
+      expect(browserHistory.push).toHaveBeenLastCalledWith({
+        pathname: '/organizations/org-slug/issues/',
+        query: {
+          cursor: undefined,
+          environment: [],
+          page: undefined,
+          project: [],
+          query: 'is:unresolved',
+          statsPeriod: '14d',
         },
-      };
-
-      wrapper.setProps({...newRouter, router: newRouter});
-      wrapper.setContext({router: newRouter});
-      wrapper.update();
-
-      wrapper.find('Button[aria-label="Pin this search"] button').simulate('click');
+      });
+    });
 
-      expect(createPin).toHaveBeenCalled();
+    it('displays a count that represents the current page', async function () {
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/issues/',
+        body: [...Array(25)].map((_, idx) => TestStubs.Group({id: `${idx}`, project})),
+        headers: {
+          'X-Hits': 500,
+          'X-Max-Hits': 1000,
+          Link: DEFAULT_LINKS_HEADER,
+        },
+      });
+      createWrapper();
 
-      await tick();
-      wrapper.update();
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
 
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/666/',
-          query: expect.objectContaining({
-            project: [123],
-            environment: ['prod'],
-            query: 'assigned:me level:fatal',
-          }),
-        })
+      expect(wrapper.getByTestId('issues-pagination')).toHaveTextContent(
```

Comment:
should we use `getByText` here?

Reply:
I've changed these, but they are kind of a pain because there are multiple `<span/>`s that wrap text and there's no native RTL way to use `findByText`, I had to use a helper function that I found from stackoverflow.